crates/tessera-backbone/src/application/path/mod.rs

```diff
@@ -15,6 +15,7 @@ pub(crate) struct PathData {
 #[async_trait]
 pub(crate) trait PathUseCase {
     async fn get_all(&self) -> Result<Vec<PathData>>;
+    async fn register(&self, path: String) -> Result<()>;
```

Comment:
Unless there's a specific requirement to own the path (e.g., storing it somewhere or performing extensive modifications), it would be more idiomatic to use a borrowed reference `&str` or `Cow<'a, str>` (This could be used more generally):  

```rust  
async fn register(&self, path: &str) -> Result<()>;
```